python/ambassador/envoy/v2/v2ready.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2021 Datawire. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License
+
+from typing import List, TYPE_CHECKING
+
+from .v2listener import V2Listener
+
+if TYPE_CHECKING:
+    from . import V2Config # pragma: no cover
+
+
+class V2Ready(dict):
+
+    @classmethod
+    def generate(cls, config: 'V2Config') -> None:
+        # Inject the ready listener to the list of listeners if enabled
+        rport = config.ir.aconf.module_lookup('ambassador', 'ready_port', -1)
+        if rport <= 0:
+            config.ir.logger.info(f"V2Ready: ==== disabled")
+            return
+
+        rip = config.ir.aconf.module_lookup('ambassador', 'ready_ip', '127.0.0.1')
```

Comment:
Kube readiness probes don't come from localhost, do they? I'm not sure this is a sensible default.

Reply:
Err, it is a sane default if the only thing we expect to use this is Emissary internally.  But if that's the case, none of this should be in user-facing toggle.s

Reply:
There are two things here:
* The idea here is to have a `secure` default (so it's not exposed in the wild by mistake).
* At some point, this endpoint might/should be consumed by the ambassador control plane, replacing the /ready check it's doing on the envoy admin listener to this one. In this case, 127.0.0.1 will be secure and a good sensible default.